ports/mimxrt/machine_sdcard.c

```diff
@@ -0,0 +1,229 @@
+    mp_buffer_info_t bufinfo;
+    mimxrt_sdcard_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    mp_get_buffer_raise(buf, &bufinfo, MP_BUFFER_WRITE);
+    // ---
+
+    if (self->initialized &&
+        sdcard_read(self, bufinfo.buf, mp_obj_get_int(block_num), bufinfo.len / SDCARD_DEFAULT_BLOCK_SIZE)) {
+        return MP_OBJ_NEW_SMALL_INT(0);
+    } else {
+        // Todo: fix upper layer instead of throwing exception here
+        mp_raise_OSError(EIO);
+    }
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_3(machine_sdcard_readblocks_obj, machine_sdcard_readblocks);
+
+// writeblocks(block_num, buf)
+STATIC mp_obj_t machine_sdcard_writeblocks(mp_obj_t self_in, mp_obj_t block_num, mp_obj_t buf) {
+    mp_buffer_info_t bufinfo;
+    mimxrt_sdcard_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    mp_get_buffer_raise(buf, &bufinfo, MP_BUFFER_WRITE);
+    // ---
+
+    if (self->initialized &&
+        sdcard_write(self, bufinfo.buf, mp_obj_get_int(block_num), bufinfo.len / SDCARD_DEFAULT_BLOCK_SIZE)) {
+        return MP_OBJ_NEW_SMALL_INT(0);
+    } else {
+        // Todo: fix upper layer instead of throwing exception here
+        mp_raise_OSError(EIO);
+    }
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_3(machine_sdcard_writeblocks_obj, machine_sdcard_writeblocks);
+
+// ioctl(op, arg)
+STATIC mp_obj_t machine_sdcard_ioctl(mp_obj_t self_in, mp_obj_t cmd_in, mp_obj_t arg_in) {
+    mimxrt_sdcard_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    mp_int_t cmd = mp_obj_get_int(cmd_in);
+
+    switch (cmd) {
+        case MP_BLOCKDEV_IOCTL_INIT: {
+            if (sdcard_detect(self)) {
+                if (sdcard_power_on(self)) {
+                    return MP_OBJ_NEW_SMALL_INT(0);
+                } else {
+                    sdcard_power_off(self);
+                    return MP_OBJ_NEW_SMALL_INT(-1);  // Initialization failed
+                }
+            } else {
+                return MP_OBJ_NEW_SMALL_INT(-1);  // Initialization failed
+            }
+        }
+        case MP_BLOCKDEV_IOCTL_DEINIT: {
+            if (sdcard_power_off(self)) {
+                return MP_OBJ_NEW_SMALL_INT(0);
+            } else {
+                return MP_OBJ_NEW_SMALL_INT(-1);  // Deinitialization failed
+            }
+        }
+        case MP_BLOCKDEV_IOCTL_SYNC: {
+            return MP_OBJ_NEW_SMALL_INT(0);
+        }
+        case MP_BLOCKDEV_IOCTL_BLOCK_COUNT: {
+            if (self->initialized) {
+                return MP_OBJ_NEW_SMALL_INT(self->block_count);
+            } else {
+                return MP_OBJ_NEW_SMALL_INT(-1);  // Card not initialized
+            }
+        }
+        case MP_BLOCKDEV_IOCTL_BLOCK_SIZE: {
+            if (self->initialized) {
+                return MP_OBJ_NEW_SMALL_INT(self->block_len);
+            } else {
+                return MP_OBJ_NEW_SMALL_INT(-1);  // Card not initialized
+            }
+        }
+        default: // unknown command
+        {
+            return MP_OBJ_NEW_SMALL_INT(-1); // error
```

Comment:
as per other ports, this should return `mp_const_none`

Reply:
fixed.